### PR TITLE
Add the Moby tool

### PR DESCRIPTION
**Pull request by justincormack**
Add the first version of the Moby tool.

See https://github.com/moby/moby/pull/32691 for context.

This tool is currently being used by LinuxKit https://github.com/linuxkit/linuxkit to assemble full system images from assemblies of containers, and willl be expanded to become a general build tool for building assemblies of components for additional platforms as well.

cute-whales

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
I expect "moby build' to work as if "docker build'

(Is there a plan to implement equivalent of "docker build"  in moby?)

File: src/cmd/moby/build.go [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/docker/moby/src/initrd"
```
Comment (ColinHebert) [inline src/cmd/moby/build.go]: moby/moby?

**Comment by tonistiigi** [comment]
What is the reason for including this tool in the same repo as the engine code(I'm avoiding calling it Moby because it seems that they are both called with the same name)?

This is a CLI utility that depends on a Docker CLI binary(not even API). It uses `docker pull` to pull images and `docker create && docker export` to extract squashed images. It also depends on tar and hypervisor launchers already in user's system but doesn't seem to depend on anything from this repo.

It also is very linux/vm centric. Even ignoring the fact that it currently extracts data to `initrd`, the format itself has terms like kernel, init, onboot that don't really make sense outside the context of creating a vm. I would even call the current tool `linuxctl`. The engine component, on the other hand, has no relation with these terms. It does not care about what os, kernel starts its process or what other tools are installed in the system. There are probably good use cases for using the engine component as one of the services inside this yml definition but that doesn't really make it different from any other software that can be assembled into a vm like this. If this tool is meant to turn into something else in the future then is there a roadmap for this?

From manifesto, I read that one of the goals is to split up more components. So it is weird that we are starting with putting 2 independent components in the same repo.

**Comment by cpuguy83** [comment]
Agreed this shouldn't live here, which I think we discussed F2F.
@justincormack Can you close if that's indeed the conclusion we came up with?

**Comment by justincormack** [comment]
The agreement we came to was that we would start by moving this to its own repo (and we will remove some of the more VM specific code, and just leave the assembly code), but the aim is that it does go here later, once its assembling stuff properly, and when the rest of the code is removed from here.

**Comment by justincormack** [comment]
This version is out of date.

**Pull request closed without merging**